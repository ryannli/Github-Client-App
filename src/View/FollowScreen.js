import React, {
	Component
} from 'react';
import {
	AppRegistry,
	StyleSheet,
	ScrollView,
	Text,
	View,
	Dimensions,
	Navigator,
	Linking,
	RefreshControl,
	AsyncStorage
} from 'react-native';

import {
	Header,
	Title,
	Button,
	Left,
	Right,
	Body,
	List,
	ListItem,
	Container,
	Content,
	Thumbnail
} from 'native-base';
import {
	SafeAreaView
} from "react-navigation"
import {
	getOwnerFollowers,
} from '../Utils/gitFollow';

import api from '../Utils/basicApi';

// Define design styles
const styles = StyleSheet.create({
	container: {
		flex: 1,
	},
	welcome: {
		fontSize: 20,
		textAlign: 'center',
		margin: 10,
	},
	instructions: {
		textAlign: 'center',
		color: '#333333',
		marginBottom: 5,
	},
	safeArea: {
		flex: 1,
		// backgroundColor: '#03A9F4',
	},
	name: {
		fontSize: 20,
		paddingLeft: 5,
		fontFamily: 'Cochin',
		fontWeight: 'bold',
	},
});

export default class FollowScreen extends Component {
	/** Render function. */
	render() {
		var followers = this.props.data;

		// Check if data is ready; otherwise, show "loading" on screen
		var follower_list = followers ? followers.map((item, index) => {
			return (
				<ListItem key={followers[index].login} onPress={() => { this.props.navigation.navigate('TabScreen', 
					{'username': followers[index].login, 
					'ownername': this.props.navigation.state.params.ownername, 
					'ownertoken': this.props.navigation.state.params.ownertoken,})}}>
				<Thumbnail size={60} source={{ uri: followers[index].avatar_url}} />
				<Body>
				<Text style={styles.name}>{followers[index].login}</Text>
			<Text note> {""} </Text>
				</Body>
				</ListItem>
			)
		}) : <Text>loading</Text>;

		return (
			<SafeAreaView style={styles.safeArea}>
			<ScrollView refreshControl={
			<RefreshControl
	            refreshing={this.props.isRefreshing}
	            onRefresh={this.props.onRefresh} />
       		 }>
			<Content style={styles.container}>
			<List>
			{follower_list}  
			</List>
			</Content>
			</ScrollView>
			</SafeAreaView>
		)
	}
}