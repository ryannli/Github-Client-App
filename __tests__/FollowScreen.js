import 'react-native';
import React from 'react';
import FollowScreen from '../src/View/FollowScreen';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

test('renders correctly', () => {
	const navigation = {
		state: {
			params: {
				username: "ryannli"
			}
		}
	};
	const tree = renderer.create(
		<FollowScreen navigation={navigation}/>
	);
	expect(tree).toMatchSnapshot();
});


test('renders correctly with refreshing', () => {
	const navigation = {
		state: {
			params: {
				username: "ryannli"
			}
		}
	};
	const refreshing = true;
	const tree = renderer.create(
		<FollowScreen navigation={navigation} refreshing={refreshing}/>
	);
	expect(tree).toMatchSnapshot();
});



test('renders correctly with data', () => {
	const navigation = {
		state: {
			params: {
				username: "ryannli"
			}
		}
	};
	const refreshing = true;
	const data = [{
		"login": "rijn",
		"id": 6976367,
		"avatar_url": "https://avatars0.githubusercontent.com/u/6976367?v=4",
		"gravatar_id": "",
		"url": "https://api.github.com/users/rijn",
		"html_url": "https://github.com/rijn",
		"followers_url": "https://api.github.com/users/rijn/followers",
		"following_url": "https://api.github.com/users/rijn/following{/other_user}",
		"gists_url": "https://api.github.com/users/rijn/gists{/gist_id}",
		"starred_url": "https://api.github.com/users/rijn/starred{/owner}{/repo}",
		"subscriptions_url": "https://api.github.com/users/rijn/subscriptions",
		"organizations_url": "https://api.github.com/users/rijn/orgs",
		"repos_url": "https://api.github.com/users/rijn/repos",
		"events_url": "https://api.github.com/users/rijn/events{/privacy}",
		"received_events_url": "https://api.github.com/users/rijn/received_events",
		"type": "User",
		"site_admin": false
	}, ]
	const tree = renderer.create(
		<FollowScreen navigation={navigation} refreshing={refreshing} data={data}/>
	);
	expect(tree).toMatchSnapshot();
});