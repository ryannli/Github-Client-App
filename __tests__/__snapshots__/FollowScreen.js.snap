// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  collapsable={undefined}
  onLayout={[Function]}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
      "paddingBottom": 0,
      "paddingLeft": 0,
      "paddingRight": 0,
      "paddingTop": 20,
    }
  }
>
  <RCTScrollView
    refreshControl={
      <RefreshControlMock
        onRefresh={undefined}
        refreshing={undefined}
      />
    }
  >
    <RCTRefreshControl />
    <View>
      <RCTScrollView
        automaticallyAdjustContentInsets={false}
        contentContainerStyle={undefined}
        contentInset={
          Object {
            "bottom": 0,
          }
        }
        enableAutoAutomaticScroll={true}
        enableResetScrollToCoords={true}
        extraHeight={75}
        extraScrollHeight={0}
        getScrollResponder={[Function]}
        handleOnScroll={[Function]}
        keyboardDismissMode="interactive"
        keyboardOpeningTime={250}
        keyboardShouldPersistTaps="handled"
        keyboardSpace={0}
        onScroll={[Function]}
        resetKeyboardSpace={[Function]}
        resetScrollToCoords={
          Object {
            "x": 0,
            "y": 0,
          }
        }
        scrollEventThrottle={1}
        scrollForExtraHeightOnAndroid={[Function]}
        scrollToEnd={[Function]}
        scrollToFocusedInput={[Function]}
        scrollToPosition={[Function]}
        showsVerticalScrollIndicator={true}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "flex": 1,
            },
            Object {
              "flex": 1,
            },
          ]
        }
        viewIsInsideTabBar={false}
        virtual={undefined}
      >
        <View>
          <View
            closeOnRowBeginSwipe={false}
            closeOnRowPress={true}
            closeOnScroll={true}
            directionalDistanceChangeThreshold={2}
            disableLeftSwipe={false}
            disableRightSwipe={false}
            leftOpenValue={0}
            previewFirstRow={false}
            recalculateHiddenLayout={false}
            rightOpenValue={0}
            style={Object {}}
            swipeToOpenPercent={50}
            virtual={undefined}
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
            >
              loading
            </Text>
          </View>
        </View>
      </RCTScrollView>
    </View>
  </RCTScrollView>
</View>
`;

exports[`renders correctly with data 1`] = `
<View
  collapsable={undefined}
  onLayout={[Function]}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
      "paddingBottom": 0,
      "paddingLeft": 0,
      "paddingRight": 0,
      "paddingTop": 20,
    }
  }
>
  <RCTScrollView
    refreshControl={
      <RefreshControlMock
        onRefresh={undefined}
        refreshing={undefined}
      />
    }
  >
    <RCTRefreshControl />
    <View>
      <RCTScrollView
        automaticallyAdjustContentInsets={false}
        contentContainerStyle={undefined}
        contentInset={
          Object {
            "bottom": 0,
          }
        }
        enableAutoAutomaticScroll={true}
        enableResetScrollToCoords={true}
        extraHeight={75}
        extraScrollHeight={0}
        getScrollResponder={[Function]}
        handleOnScroll={[Function]}
        keyboardDismissMode="interactive"
        keyboardOpeningTime={250}
        keyboardShouldPersistTaps="handled"
        keyboardSpace={0}
        onScroll={[Function]}
        resetKeyboardSpace={[Function]}
        resetScrollToCoords={
          Object {
            "x": 0,
            "y": 0,
          }
        }
        scrollEventThrottle={1}
        scrollForExtraHeightOnAndroid={[Function]}
        scrollToEnd={[Function]}
        scrollToFocusedInput={[Function]}
        scrollToPosition={[Function]}
        showsVerticalScrollIndicator={true}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "flex": 1,
            },
            Object {
              "flex": 1,
            },
          ]
        }
        viewIsInsideTabBar={false}
        virtual={undefined}
      >
        <View>
          <View
            closeOnRowBeginSwipe={false}
            closeOnRowPress={true}
            closeOnScroll={true}
            directionalDistanceChangeThreshold={2}
            disableLeftSwipe={false}
            disableRightSwipe={false}
            leftOpenValue={0}
            previewFirstRow={false}
            recalculateHiddenLayout={false}
            rightOpenValue={0}
            style={Object {}}
            swipeToOpenPercent={50}
            virtual={undefined}
          >
            <View
              accessibilityComponentType={undefined}
              accessibilityLabel={undefined}
              accessibilityTraits={undefined}
              accessible={true}
              hasTVPreferredFocus={undefined}
              hitSlop={undefined}
              isTVSelectable={true}
              nativeID={undefined}
              onLayout={undefined}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={null}
              testID={undefined}
              tvParallaxProperties={undefined}
            >
              <View
                onPress={[Function]}
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "transparent",
                    "borderBottomWidth": 0.5,
                    "borderColor": "#c9c9c9",
                    "flexDirection": "row",
                    "marginLeft": 15,
                    "paddingBottom": 13,
                    "paddingRight": 15,
                    "paddingTop": 13,
                  }
                }
                virtual={undefined}
              >
                <Image
                  size={60}
                  source={
                    Object {
                      "uri": "https://avatars0.githubusercontent.com/u/6976367?v=4",
                    }
                  }
                  style={
                    Object {
                      "borderRadius": 28,
                      "height": 56,
                      "width": 56,
                    }
                  }
                  virtual={undefined}
                />
                <View
                  style={
                    Object {
                      "alignItems": null,
                      "alignSelf": null,
                      "flex": 1,
                    }
                  }
                  virtual={undefined}
                >
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Object {
                        "fontFamily": "Cochin",
                        "fontSize": 20,
                        "fontWeight": "bold",
                        "paddingLeft": 5,
                      }
                    }
                  >
                    rijn
                  </Text>
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    note={true}
                  >
                     
                    
                     
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
      </RCTScrollView>
    </View>
  </RCTScrollView>
</View>
`;

exports[`renders correctly with refreshing 1`] = `
<View
  collapsable={undefined}
  onLayout={[Function]}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
      "paddingBottom": 0,
      "paddingLeft": 0,
      "paddingRight": 0,
      "paddingTop": 20,
    }
  }
>
  <RCTScrollView
    refreshControl={
      <RefreshControlMock
        onRefresh={undefined}
        refreshing={undefined}
      />
    }
  >
    <RCTRefreshControl />
    <View>
      <RCTScrollView
        automaticallyAdjustContentInsets={false}
        contentContainerStyle={undefined}
        contentInset={
          Object {
            "bottom": 0,
          }
        }
        enableAutoAutomaticScroll={true}
        enableResetScrollToCoords={true}
        extraHeight={75}
        extraScrollHeight={0}
        getScrollResponder={[Function]}
        handleOnScroll={[Function]}
        keyboardDismissMode="interactive"
        keyboardOpeningTime={250}
        keyboardShouldPersistTaps="handled"
        keyboardSpace={0}
        onScroll={[Function]}
        resetKeyboardSpace={[Function]}
        resetScrollToCoords={
          Object {
            "x": 0,
            "y": 0,
          }
        }
        scrollEventThrottle={1}
        scrollForExtraHeightOnAndroid={[Function]}
        scrollToEnd={[Function]}
        scrollToFocusedInput={[Function]}
        scrollToPosition={[Function]}
        showsVerticalScrollIndicator={true}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "flex": 1,
            },
            Object {
              "flex": 1,
            },
          ]
        }
        viewIsInsideTabBar={false}
        virtual={undefined}
      >
        <View>
          <View
            closeOnRowBeginSwipe={false}
            closeOnRowPress={true}
            closeOnScroll={true}
            directionalDistanceChangeThreshold={2}
            disableLeftSwipe={false}
            disableRightSwipe={false}
            leftOpenValue={0}
            previewFirstRow={false}
            recalculateHiddenLayout={false}
            rightOpenValue={0}
            style={Object {}}
            swipeToOpenPercent={50}
            virtual={undefined}
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
            >
              loading
            </Text>
          </View>
        </View>
      </RCTScrollView>
    </View>
  </RCTScrollView>
</View>
`;
